usagov/usagov-2021
# Usagov 1546 ability to add site banner

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1546

## Description

This PR adds a new feature that allows Content Administrators to add a banner to the `header_top` region of the site.
Here is a summary of all the things that were done:
- Update permissions to allow Content Admins to have access to the page where blocks are managed
- Create a new block_content type called site_banner, which uses the USWDS Alert component
- Update permissions to allow Content Admins to create, edit and delete site_banners (they do not have access to any other type of block)
- Added a checkbox to the site_banner form to allow the content admin to place and remove a banner from the header_top region.
- Added a new hook to the usa_twig_vars.module and added a twig template for the banner placement
- I wrote a readme in the repo for developers that explains how this feature works and all the files related to it.
- Updated the design of these alerts to make them look better.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Run `bin/init` and `bin/drush cim`
2. Create a Content Administrator account and log in using that account
3. Go to the Content blocks page `localhost/admin/content/block`
4. Click `+ Add content block`.
5. Fill in the required fields for the Site Banner.
   - Make sure you select the correct language for the banner.
6. Find the checkbox labeled `Place above header`.
   - Check this box if you want the banner to appear in the `header_top` region.
   - Uncheck this box if you want to remove the banner from the `header_top` region.
7. Click `Save` to create or update the Site Banner block.
8. Visit the site to ensure the banner appears or is removed from the region.
9. Follow the previous steps to create a banner for each Alert Status and both languages. Also, check that it looks like this:

Desktop:
<img width="806" alt="image (1)" src="https://github.com/usagov/usagov-2021/assets/39020809/51d90d62-db74-4bed-85ac-9c60ed040b69">

Tablet:
<img width="645" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/6a5bf069-8e48-41f6-9c33-7989a64baf4c">

Mobile:
![image](https://github.com/usagov/usagov-2021/assets/39020809/7f10a4aa-e538-4a53-a742-f100c2fd58b8)


10. Make sure content admins can only edit this type of block.
11. Make sure to read the code and documentation. Please let me know if there is anything that can be improved.

### Requires New Config

### Requires New Content

### Validation Steps

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
